IPython/nbconvert/filters/highlight.py

```diff
@@ -113,7 +113,7 @@ def _pygments_highlight(source, output_formatter, language='ipython', metadata=N
 
     # If the cell uses a magic extension language,
     # use the magic language instead.
-    if language == 'ipython' \
+    if language in ('ipython', 'ipython3') \
```

Comment:
I am hesitant to make more changes than necessary to fix the bug.  It is possible that in the cases where `python2` is passed in as language, it should not be replaced with the magics language.